tests/e2e/homework/tests/passengerDetails.spec.ts

```diff
@@ -0,0 +1,35 @@
+import { Login } from '../components/loginPage';
+import {test, expect} from '@playwright/test';
+import { FlightPage } from '../components/flightPage';
+import { PassengerDetails } from '../components/passengerDetails';
+
+let login: Login;
+let flightPage: FlightPage;
+test.beforeEach('go to page', async({page})=>{
+    login = new Login (page);
+    flightPage = new FlightPage (page);
+    await login.goToPage();
+    await login.login('agileway', 'testW1se');
+    await flightPage.flightSelection('New York', 'Sydney', '30', 'July 2025');
+})
+
+test('valid Passenger Details', async({page})=>{
+    const passengerDetails = new PassengerDetails(page);
+    await passengerDetails.passengerDetailsInput('Test', 'Testaitis');
+    await expect(page.locator('h2')).toBeVisible();
+})
+
+test.describe('invalid Passenger Details', () => {    
+
+test('no input added', async ({page}) => {
+    const passengerDetails = new PassengerDetails(page);
+    await passengerDetails.passengerDetailsInput('', '');
+    await expect(page.locator('#flash_alert')).toBeVisible(); 
```

Comment:
Please extract common components, such as alerts and their validation to basePage class (then extend existing page classes with this new class). Please adjust throughout the solution